lddwrap/__init__.py

```diff
@@ -0,0 +1,228 @@
+
+        return Dependency(soname=None, path=pathlib.Path(parts[0]), found=True, mem_address=parts[1])
+
+
+def get_dependencies(path: pathlib.Path,
+                     verbose: bool = False,
+                     unused: bool = False,
+                     env: Optional[Dict[str, str]] = None) -> List[Dependency]:
+    """
+    Retrieve a list of dependencies of the given binary.
+
+    :param path: path to a file
+    :param verbose: additionally print all information
+    :param unused: additional adds information if dependency is used
+    :param env: the environment to use
+    :return: list of dependencies
+
+    Options copied from ldd documentary:
+    --version
+              Print the version number of ldd.
+
+       -v, --verbose
+              Print all information, including, for example, symbol
+              versioning information.
+
+       -u, --unused
+              Print unused direct dependencies.  (Since glibc 2.3.4.)
+
+       -d, --data-relocs (not implemented)
+              Perform relocations and report any missing objects (ELF only).
+
+       -r, --function-relocs (not implemented)
+              Perform relocations for both data objects and functions, and
+              report any missing objects or functions (ELF only).
+
+       --help Usage information. (not implemented)
+    """
+    commands = ["ldd"]  # type: List[str]
+    if not path.is_file():
+        raise IOError("Not a file: {}".format(path))
+
+    if verbose and unused:
+        get_dependencies(path=path, verbose=True)
+        verbose = False
+
+    if verbose:
+        commands.append("--verbose")
+
+    commands.append(path.as_posix())
+
+    proc = subprocess.Popen(commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=env)
+
+    out, err = proc.communicate()
+    if proc.returncode != 0:
+        raise RuntimeError("Failed ldd-ing external libraries of {} with code {}:\nout:\n{}\n\nerr:\n{}".format(
+            path, proc.returncode, out, err))
+
+    out_unused = ""
+    if unused:
+        commands = ["ldd", "--unused", path.as_posix()]
+        proc_unused = subprocess.Popen(
+            commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=env)
+        out_unused, err_unused = proc_unused.communicate()
+        # return code = 0 -> no unused dependencies, return code = 1 -> some unused dependencies
+        if proc_unused.returncode not in [0, 1]:
+            raise RuntimeError("Failed ldd-ing external libraries of {} with code {}:\nout:\n{}\n\nerr:\n{}".format(
+                path, proc.returncode, out_unused, err_unused))
+
+    dependencies = dependencies_parse_helper(
+        out=out, out_unused=out_unused, verbose=verbose, unused=unused)  # type: List[Dependency]
+
+    return dependencies
+
+
+def dependencies_parse_helper(out: str, out_unused: str, verbose: Optional[bool] = None,
```

Comment:
this function would be refractored and split to two after changes in `get_dependencies` (see Comment C)